onnxruntime/core/platform/posix/hardware_core_enumerator_linux.cc

```diff
@@ -0,0 +1,51 @@
+#include "hardware_core_enumerator_linux.h"
+
+#ifdef __linux__
+#include <cpuid.h>
+#include <vector>
+
+namespace onnxruntime {
+
+IntelChecks CheckIntel() {
+  IntelChecks intel_check = {false, false};
+  bool isIntel_SpecifiedPlatform = false;
+  unsigned int regs[4] = {0};
+  __get_cpuid(0, &regs[0], &regs[1], &regs[2], &regs[3]);
+
+  const unsigned int kVendorID_Intel[3] = {0x756e6547, 0x6c65746e, 0x49656e69};  // "GenuntelineI"
+  bool is_Intel = (regs[1] == kVendorID_Intel[0] &&
+                  regs[2] == kVendorID_Intel[1] &&
+                  regs[3] == kVendorID_Intel[2]);
+  if (!is_Intel) {
+    return intel_check;  // if not an Intel CPU, return early
+  }
+
+  __get_cpuid(1, &regs[0], &regs[1], &regs[2], &regs[3]);
+
+  unsigned int base_family = (regs[0] >> 8) & 0xF;
+  unsigned int base_model = (regs[0] >> 4) & 0xF;
+  unsigned int extended_model = (regs[0] >> 16) & 0xF;
+
+  unsigned int model = (base_family == 0x6 || base_family == 0xF) ? (base_model + (extended_model << 4)) : base_model;
+
+  const std::vector<unsigned int> kVendorID_IntelSpecifiedPlatformIDs = {
```

Comment:
don't need to use a vector here
```suggestion
  const unsigned int kVendorID_IntelSpecifiedPlatformIDs[3] = {
```

Reply:
Hey @edgchen1 the reason why a vector is used here rather than an array is so that `kVendorID_IntelSpecifiedPlatformIDs` can be expanded on for future platform releases, so here it makes sense to use a vector over an array.

Reply:
a vector makes sense if the collection is dynamic at runtime. here, it seems like it could be a compile-time constant (`constexpr` might be good too).

I think that this would also work, and make it easier to update the code with additional entries later on:
```suggestion
  const unsigned int kVendorID_IntelSpecifiedPlatformIDs[] = {
```